domain/application/state/state.go

```diff
@@ -21,22 +19,10 @@ import (
 // Composes both application and charm state, so we can interact with both
 // from the single state, whilst also keeping the concerns separate.
 type State struct {
-	*domain.StateBase
 	*ApplicationState
 	*CharmState
 }
 
-// NewState returns a new State for interacting with the underlying state.
-func NewState(factory database.TxnRunnerFactory, logger logger.Logger) *State {
-	base := &commonStateBase{
-		StateBase: domain.NewStateBase(factory),
-	}
-	return &State{
-		ApplicationState: NewApplicationState(base, logger),
-		CharmState:       NewCharmState(base),
-	}
-}
```

Comment:
I am just wondering if this is such a good idea or not. In theory, I like that we don't have one big state. The service can compose multiple states to get the desired state, but it does mean that we have multiple states instead of one.

I think I can live with this, we can always switch it back if we so desire.

Reply:
If I didn't do this, it wouldn't compile due to an ambiguous run atomic reference - can't recall the exact error now. But I think I like keeping things explicitly separate TBH.